aws/resource_aws_ecs_service.go

```diff
@@ -287,21 +291,12 @@ func resourceAwsEcsServiceCreate(d *schema.ResourceData, meta interface{}) error
 		out, err = conn.CreateService(&input)
 
 		if err != nil {
-			awsErr, ok := err.(awserr.Error)
-			if !ok {
-				return resource.NonRetryableError(err)
-			}
-			if awsErr.Code() == "InvalidParameterException" {
-				log.Printf("[DEBUG] Trying to create ECS service again: %q",
-					awsErr.Message())
+			if isAWSErr(err, ecs.ErrCodeClusterNotFoundException, "") {
 				return resource.RetryableError(err)
 			}
-			if awsErr.Code() == "ClusterNotFoundException" {
-				log.Printf("[DEBUG] Trying to create ECS service again: %q",
-					awsErr.Message())
+			if isAWSErr(err, ecs.ErrCodeInvalidParameterException, "Please verify that the ECS service role being passed has the proper permissions.") {
```

Comment:
I trust you that you ran all ECS related acceptance tests a couple of times to verify that we're not missing any other message/format here. 😃